airbyte-integrations/connectors/source-kyve/source_kyve/stream.py

```diff
@@ -55,22 +56,45 @@ def __init__(self, config: Mapping[str, Any], pool_data: Mapping[str, Any], **kw
         self.runtime = pool_data.get("runtime")
 
         self.url_base = config["url_base"]
-        # this is an ugly solution but has to parsed by source to be a single item
+
         self._offset = int(config["start_ids"])
 
-        self.page_size = config["page_size"]
+        self._start_key = int(config["start_keys"])
+        self._end_key = int(config["end_keys"])
+
+        self._tendermint_normalization = config["tendermint_normalization"]
+
+        self._reached_end = False
+
+        self.page_size = 100
         self.max_pages = config.get("max_pages", None)
+
         # For incremental querying
-        self._cursor_value = None
+        self._cursor_value = self._offset
 
     def get_json_schema(self) -> Mapping[str, Any]:
-        # This is KYVE's default schema and won't be changed.
-        schema = {
-            "$schema": "http://json-schema.org/draft-04/schema#",
-            "type": "object",
-            "properties": {"key": {"type": "string"}, "value": {"type": "object"}},
-            "required": ["key", "value"],
-        }
+        if self._tendermint_normalization:
+            # This is KYVE's default schema and won't be changed.
+            schema = {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                    "height": {"type": "integer"},
+                    "value": {"type": "any"},
+                    "type": {"type": "string"},
+                    "array_index": {"type": "integer"},
+                    "offset": {"type": "string"},
+                },
+                "required": ["key", "value"],
+            }
+        else:
+            # This is KYVE's default schema and won't be changed.
+            schema = {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {"key": {"type": "string"}, "value": {"type": "any"}, "offset": {"type": "string"}},
```

Comment:
Again, or you build a parser to map to the correct value type of cast to string a default.